### DISCUSSION TITLE
RA/DEC (J.2000) Values Displayed by Stellarium.

### DISCUSSION THREAD

**Discussion by BikramAjay** [Stellarium/stellarium#1782] [General]
In the light of astrometric transformations shown in figure 1(see attached image from http://www.iausofa.org), which one (out of ICRS, BCRS, GCRS, CIRS, TIRS, ITRS)  is the  RA/DEC (J.2000) values displayed by Stellarium. 
![Screenshot 2021-07-01 140859](https://user-images.githubusercontent.com/86772525/124171319-66c33f80-da76-11eb-86c1-17ada353527f.jpg)
![Screenshot 2021-07-01 045243](https://user-images.githubusercontent.com/86772525/124171331-6b87f380-da76-11eb-9d63-2d2e7def4e70.jpg)

**Comment by gzotti** [comment]
None. Stellarium is predating and therefore not using SOFA but is more based on the classical 'Astronomical Algorithms' from Jean Meeus, and sources described in the Stellarium User Guide and Stellarium's source code documentation. The J2000 coordinates for stars should closely resemble ICRS, but currently we cannot give any guarantees. 

We hope to finally have annual aberration implemented later this year. Parallax is quasi automatic by vector math of observer locations. I have currently no intention to implement diurnal aberration, nor Solar light deflection, nor polar motion.  

If you want to implement the most up-to-date standards or even introduce rigorous tests that could ascertain accurate implementation of all steps, please feel free to do so. In this respect, Stellarium-web is the more modern project.

**Reply by BikramAjay** [reply]
Thanks for the reply.

**Reply by BikramAjay** [reply]
Hi Mr Zotti,
                  Do you know which file (in the src folder of Stellarium source code) implements epoch transformation of the stars? I mean when I change the year field of date GUI, say from 2021 to 1000AD, which file implements new star position (RA and DEC) calculations?
                  Thanks in advance.

**Reply by gzotti** [reply]
Look into ZoneData and ZoneArray. And Appendix C of the User Guide. This part of the project has AFAICS not seen significant updates in the past 10 years or so. Good luck.

**Reply by BikramAjay** [reply]
Thanks
